### PR TITLE
run tests on GitHub Actions

### PR DESCRIPTION
**Pull request by leque**
This PR is a reboot of #96 and #197. This runs `dune runtest` and then build some documents by using installed SATySFi on the CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,3 +51,18 @@ jobs:
...
         run: opam exec -- make all
+
+      - name: Run tests
+        run: |
+          opam exec -- dune runtest
+
+      - name: Install SATySFi
+        run: |
+          opam install -y .
+          ./download-fonts.sh
```
Comment (na4zagin3) [inline .github/workflows/ci.yml:62]: Can you cache the downloaded fonts (see https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows)?  The other option is to have OPAM cache downloaded files (#149).

When I was using Travis CI, running download-fonts.sh without caching had the CI instances banned from some font distributing sites.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -51,3 +51,18 @@ jobs:
...
+          sudo ./install-libs.sh
```
Comment (na4zagin3) [inline .github/workflows/ci.yml:63]: This is a matter of taste.  You can install the lib files under `~/.satysfi` instead.

```
./install-libs.sh ~/.satysfi
```

Comment (leque) [reply .github/workflows/ci.yml]: fair enough!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (leque) [reply .github/workflows/ci.yml]: Thank you. I have added cache settings.

**Comment by leque** [comment]
Updated. Now we install fonts into `~/.satysfi`, and use caches to avoid downloading fonts on each build.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (na4zagin3) [reply .github/workflows/ci.yml]: Thank you!

**Comment by gfngfn** [comment]
Sorry for the late response… I understand that caching of fonts is enabled and that it seems indeed working. Thank you for the improvement!

**Pull request merged**